node/src/accountant/db_access_objects/utils.rs

```diff
@@ -303,27 +304,34 @@ impl<T: Copy> From<&RangeQuery<T>> for CustomQuery<T> {
     }
 }
 
-pub fn remap_payable_accounts(accounts: Vec<PayableAccount>) -> Vec<UiPayableAccount> {
+#[derive(Debug, PartialEq, Eq)]
+pub struct PayableAccountWithTxInfo {
+    pub account: PayableAccount,
+    pub tx_opt: Option<TxProcessingInfo>,
+}
+
+pub fn remap_payable_accounts(accounts: Vec<PayableAccountWithTxInfo>) -> Vec<UiPayableAccount> {
     accounts
         .into_iter()
-        .map(|account| UiPayableAccount {
-            wallet: account.wallet.to_string(),
-            age_s: to_age(account.last_paid_timestamp),
-            balance_gwei: {
-                let gwei = (account.balance_wei / (WEIS_IN_GWEI as u128)) as u64;
-                if gwei > 0 {
-                    gwei
-                } else {
-                    panic!(
-                        "Broken code: PayableAccount with less than 1 gwei passed through db query \
-                         constraints; wallet: {}, balance: {}",
-                        account.wallet, account.balance_wei
-                    )
-                }
-            },
-            pending_payable_hash_opt: account
-                .pending_payable_opt
-                .map(|full_id| full_id.hash.to_string()),
+        .map(|account_with_tx_info| {
+            let account = account_with_tx_info.account;
+            UiPayableAccount {
+                wallet: account.wallet.to_string(),
+                age_s: to_age(account.last_paid_timestamp),
+                balance_gwei: {
+                    let gwei = (account.balance_wei / (WEIS_IN_GWEI as u128)) as u64;
+                    if gwei > 0 {
+                        gwei
+                    } else {
+                        panic!(
```

Comment:
This panic could be abstracted if you use a `TryFrom` conversion here, it'll look cleaner if you do that way.